### PR TITLE
Read host os-release file when running inside the snap

### PR DESCRIPTION
**Pull request by morphis**

### CODE REVIEW

File: src/anbox/cmds/system_info.cpp [hunk 1]
```diff
@@ -94,14 +95,18 @@ class SystemInformation {
  private:
   void collect_os_info() {
     os_info_.snap_based = (getenv("SNAP") != nullptr);
```
Comment (vosst) [inline src/anbox/cmds/system_info.cpp]: While you are at it, you could replace the naked getenv here.

File: src/anbox/cmds/system_info.cpp [hunk 2]
```diff
@@ -94,14 +95,18 @@ class SystemInformation {
...
+    // If we're running from within a snap the best we can do is to
+    // access the hostfs and read the os-release file from there.
+    if (os_info_.snap_based && fs::exists(host_os_release_path))
+        path = host_os_release_path;
+
+    // Double check that there aren't any permission errors when trying
+    // to access the file (e.g. because of snap confinement)
+    if (fs::exists(path)) {
       anbox::utils::EnvironmentFile os_release(os_release_path);
       os_info_.name = os_release.value("NAME");
```
Comment (vosst) [inline src/anbox/cmds/system_info.cpp]: Please define constants for the magic keys here.

File: src/anbox/cmds/system_info.cpp [hunk 3]
```diff
@@ -94,14 +95,18 @@ class SystemInformation {
...
-    if (fs::exists(os_release_path)) {
+    fs::path path = os_release_path;
+    // If we're running from within a snap the best we can do is to
+    // access the hostfs and read the os-release file from there.
+    if (os_info_.snap_based && fs::exists(host_os_release_path))
+        path = host_os_release_path;
+
+    // Double check that there aren't any permission errors when trying
+    // to access the file (e.g. because of snap confinement)
+    if (fs::exists(path)) {
```
Comment (vosst) [inline src/anbox/cmds/system_info.cpp]: How do you handle the case of being unable to read the respective file?

File: src/anbox/cmds/system_info.cpp [hunk 4]
```diff
@@ -94,14 +95,18 @@ class SystemInformation {
...
       anbox::utils::EnvironmentFile os_release(os_release_path);
```
Comment (vosst) [inline src/anbox/cmds/system_info.cpp]: This should be `path` now.

File: src/anbox/cmds/system_info.cpp (hunk 3 above)
Comment (morphis) [inline src/anbox/cmds/system_info.cpp]: In that case the default value "n/a" will be used which is assigned in the struct below.

File: src/anbox/cmds/system_info.cpp (hunk 4 above)
Comment (morphis) [inline src/anbox/cmds/system_info.cpp]: Done.

File: src/anbox/cmds/system_info.cpp (hunk 1 above)
Comment (morphis) [inline src/anbox/cmds/system_info.cpp]: Done.

File: src/anbox/cmds/system_info.cpp (hunk 2 above)
Comment (morphis) [inline src/anbox/cmds/system_info.cpp]: Done.

**Pull request merged**